argOder failed to handle function calls inside known call expressions | false positive

argOder failed to handle function calls inside known call expressions

```
cmd/agentd/main.go:174:25: argOrder: probably meant `strings.TrimSuffix(filepath.Ext(configFileName), configFileName)` (gocritic)
			config.SetConfigName(strings.TrimSuffix(configFileName, filepath.Ext(configFileName)))
```

there is no error in original call:
```
config.SetConfigName(strings.TrimSuffix(configFileName, filepath.Ext(configFileName)))
```

--- a/checkers/argOrder_checker.go
+++ b/checkers/argOrder_checker.go
@@ -59,12 +59,11 @@ func (c *argOrderChecker) VisitExpr(expr ast.Expr) {
 }
 
 func (c *argOrderChecker) isConstLiteral(x ast.Expr) bool {
-	if c.ctx.TypesInfo.Types[x].Value != nil {
-		return true
-	}
-
 	// Also permit byte slices.
 	switch x := x.(type) {
+	case *ast.BasicLit:
+		return true
+
 	case *ast.CallExpr:
 		// Handle `[]byte("abc")` as well.
 		if len(x.Args) != 1 || !astp.IsBasicLit(x.Args[0]) {

--- a/checkers/testdata/argOrder/negative_tests.go
+++ b/checkers/testdata/argOrder/negative_tests.go
@@ -2,6 +2,7 @@ package checker_test
 
 import (
 	"bytes"
+	"path/filepath"
 	"strings"
 )
 
@@ -40,4 +41,10 @@ func properArgsOrder(s string, b []byte) {
 	_ = bytes.TrimSuffix(b, []byte(":"))
 	_ = strings.Split(s, "/")
 	_ = bytes.Split(b, []byte("/"))
+}
+
+func trimSuffix() {
+	// See #963.
+	const configFileName = "foo.json"
+	_ = strings.TrimSuffix(configFileName, filepath.Ext(configFileName))
 }

